atd-etl/prefect/afd_ems_incidents/ems_incident_upload.py

```diff
@@ -0,0 +1,440 @@
+    EMS_S3_SOURCE_PREFIX = environment_variables["EMS_S3_SOURCE_PREFIX"]
+    EMS_S3_ARCHIVE_PREFIX = environment_variables["EMS_S3_ARCHIVE_PREFIX"]
+else:
+    DB_USERNAME = os.getenv("AFD_DB_USERNAME")
+    DB_PASSWORD = os.getenv("AFD_DB_PASSWORD")
+    DB_HOSTNAME = os.getenv("AFD_DB_HOSTNAME")
+    DB_PORT = os.getenv("AFD_DB_PORT")
+    DB_DATABASE = os.getenv("AFD_DB_DATABASE")
+    AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
+    AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
+    AFD_S3_SOURCE_BUCKET = os.getenv("AFD_S3_SOURCE_BUCKET")
+    AFD_S3_ARCHIVE_BUCKET = os.getenv("AFD_S3_ARCHIVE_BUCKET")
+    AFD_S3_SOURCE_PREFIX = os.getenv("AFD_S3_SOURCE_PREFIX")
+    AFD_S3_ARCHIVE_PREFIX = os.getenv("AFD_S3_ARCHIVE_PREFIX")
+    EMS_S3_SOURCE_BUCKET = os.getenv("EMS_S3_SOURCE_BUCKET")
+    EMS_S3_ARCHIVE_BUCKET = os.getenv("EMS_S3_ARCHIVE_BUCKET")
+    EMS_S3_SOURCE_PREFIX = os.getenv("EMS_S3_SOURCE_PREFIX")
+    EMS_S3_ARCHIVE_PREFIX = os.getenv("EMS_S3_ARCHIVE_PREFIX")
+
+
+
+@task
+def get_timestamp():
+    current = datetime.now()
+    return time.mktime(current.timetuple())
+
+
+@task
+def get_most_recent_email():
+    """
+    Find the most recently updated file in the bucket. This will be the newest email.
+    :return: string
+    """
+
+    session = boto3.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
+    s3 = session.resource("s3")
+
+    bucket = s3.Bucket(EMS_S3_SOURCE_BUCKET)
+
+    files = []
+    for file in bucket.objects.filter(Prefix=EMS_S3_SOURCE_PREFIX):
+        files.append(file)
+    files.sort(key=lambda x: x.last_modified, reverse=True)
+
+    file = files[0].get()
+    email = file["Body"].read().decode("utf-8")
+
+    return email
+
+
+@task
+def extract_email_attachment(message):
+    # Given the s3 object content is the SES email,
+    # get the message content and attachment using email package
+    msg = email.message_from_string(message)
+    attachment = msg.get_payload()[1]
+
+    tmpdir = tempfile.mkdtemp()
+
+    logger = prefect.context.get("logger")
+    logger.info(f"Tmpdir: {tmpdir}")
+
+    # Write the attachment to a temp location
+    open(f"{tmpdir}/attachment.csv", "wb").write(attachment.get_payload(decode=True))
+    return tmpdir
+
+
+@task
+def upload_attachment_to_S3(location, timestamp):
+    session = boto3.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
+    s3 = session.resource("s3")
+
+    s3.Bucket(EMS_S3_ARCHIVE_BUCKET).upload_file(
+        f"{location}/attachment.csv",
+        f"{EMS_S3_ARCHIVE_PREFIX}upload-{timestamp}.csv",
+    )
+    return True
+
+
+@task
+def create_and_parse_dataframe(location):
+    # Parse CSV into pandas dataframe
+    data = pandas.read_csv(f"{location}/attachment.csv", header=0, on_bad_lines='skip',  encoding='ISO-8859-1')
```

Comment:
Is it a big deal that we could be losing some data here with `on_bad_lines='skip'`? Based on Pandas docs it looks like it's probably dropping rows that have too many commas and therefore columns?

Reply:
You're 100% right, yea, this will cause a records to be skipped if they are malformed. My experience is that ~ 1 / 20,000 records from EMS are malformed, and it's not really obvious if there is a way to repair them. 

That said, the low rate of malformed records makes me think this is an OK concession. We could emit something like a warning to make them more visible in Prefect, but I'd want to spin that off as a new issue. @patrickm02L